src/modules/auth/hooks/useRestrictWindowActions.js

```diff
@@ -0,0 +1,14 @@
+import { useCallback, useEffect } from 'react';
+
+export const useRestrictWindowActions = ({ actions, window }) => {
+  const restrictAction = useCallback((event) => {
+    event.preventDefault();
+  }, []);
+
+  useEffect(() => {
+    actions?.forEach((action) => action && window.addEventListener(action, restrictAction));
+    return () => {
+      actions?.forEach((action) => action && window.removeEventListener(action, restrictAction));
+    };
+  }, [actions, window, restrictAction]);
+};
```

Comment:
do we need to pass in window as an argument to the hook?. Since `window` is a global variable and available any where we should be able to get it in the hook without passing it to the hook right ?